wagtail/tests/test_page_model.py

```diff
@@ -2141,6 +2142,41 @@ def test_copy_alias_page(self):
         # The copy should just be a copy of the original page, not an alias
         self.assertIsNone(about_us_alias_copy.alias_of)
 
+    def test_copy_page_copies_restriction_when_parent_have_no_restriction(self):
+        """ Test case where Page we are copy from have restriction and parent have no restriction """
+
+        homepage = Page.objects.get(url_path='/home/')
+        child_page_1 = SimplePage(title="Child Page 1", slug="child-page-1", content="hello child page 1")
+        homepage.add_child(instance=child_page_1)
+
+        # Add PageViewRestriction to child_page_1
+        PageViewRestriction.objects.create(page=child_page_1, password='hello')
+        child_page_2 = child_page_1.copy(update_attrs={'title': "Child Page 2", 'slug': 'child-page-2'})
+
+        # check that the homepage is not in the result
+        self.assertFalse(PageViewRestriction.objects.filter(page=homepage).exists())
+
+        # check that the child_page_2 is in the result
+        self.assertTrue(PageViewRestriction.objects.filter(page=child_page_2).exists())
+
+    def test_copy_page_not_copies_restriction_when_parent_have_restriction(self):
+        """ Test case where Page we are copy from have restriction and parent have  restriction"""
+
+        homepage = Page.objects.get(url_path='/home/')
+        child_page_1 = SimplePage(title="Child Page 1", slug="child-page-1", content="hello child page 1")
+        homepage.add_child(instance=child_page_1)
```

Comment:
This test is a bit misleading because the same parent page is being used as the source and destination. As per @zerolab's [original specification](https://github.com/wagtail/wagtail/issues/6212#issuecomment-655377744), there are three different situations to consider when copying page X from parent A to parent B, which I think should become three distinct tests here:

1. Page X has a restriction, but neither A nor B have one; page X's restriction should be copied
2. Page A has a restriction; this should not be copied over when copying page X
3. Page X has a restriction, and page B has a restriction; in this case page X's restriction should not be copied over (because then the resulting copy would have multiple restrictions applied to it).

As it stands, the test above is testing situation 1, and this one is testing a mixture of 2 and 3.